mdata/cache/if.go

```diff
@@ -7,16 +7,23 @@ import (
 )
 
 type Cache interface {
-	Add(string, uint32, chunk.IterGen)
+	Add(string, string, uint32, chunk.IterGen)
 	CacheIfHot(string, uint32, chunk.IterGen)
 	Stop()
 	Search(context.Context, string, uint32, uint32) *CCSearchResult
+	DelMetric(string) *CCDelMetricResult
+	Reset() *CCDelMetricResult
```

Comment:
pointer to CCDelMetricResult is overkill in both cases

Reply:
https://github.com/grafana/metrictank/pull/555/commits/3abd68cdf4aa473efc5885edea86176ac6e5ac01